CMakeLists.txt

```diff
@@ -320,15 +321,9 @@ if (DALI_CLANG_ONLY)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unknown-cuda-version --cuda-path=${CUDA_TOOLKIT_PATH_VAR} ${CUDA_gencode_flags_clang}")
 endif()
 
-# Add ptx & bin flags for cuda compiler (nvcc)
-if(USE_CMAKE_CUDA_ARCHITECTURES)
-  CUDA_get_cmake_cuda_archs(CMAKE_CUDA_ARCHITECTURES ${CUDA_targeted_archs})
-  message(STATUS "Generated CMAKE_CUDA_ARCHITECTURES: ${CMAKE_CUDA_ARCHITECTURES}")
-else()
-  CUDA_get_gencode_args(CUDA_gencode_flags_nvcc ${CMAKE_CUDA_COMPILER} ${CUDA_targeted_archs})
-  message(STATUS "Generated gencode flags: ${CUDA_gencode_flags_nvcc}")
-  set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} ${CUDA_gencode_flags_nvcc}")
-endif()
+CUDA_get_gencode_args(CUDA_gencode_flags_nvcc ${CMAKE_CUDA_COMPILER} ${CUDA_targeted_archs})
+message(STATUS "Generated gencode flags: ${CUDA_gencode_flags_nvcc}")
+set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} ${CUDA_gencode_flags_nvcc}")
```

Comment:
Now it can have both the gencode flags generated manually and set via CMAKE_CUDA_FLAGS + the ones specified by the user in CMAKE_CUDA_ARCHITECTURES and handled by CMake.
Plus in our functions we don't support the same syntax as CMake (only numeric values, and not flags like `70-real`). 
The USE_CMAKE_CUDA_ARCHITECTURES was intended as a way to keep our build config and test/convert the CMAKE_CUDA_ARCHITECTURES (I think the earliest CMake version that we currently allow doesn't support CMAKE_CUDA_ARCHITECTURES).

I would prefer to do as suggested earlier, so disallow setting both USE_CMAKE_CUDA_ARCHITECTURES and CMAKE_CUDA_ARCHITECTURES, with an option to use CMAKE_CUDA_ARCHITECTURES if provided. Maybe if setting USE_CMAKE_CUDA_ARCHITECTURES to `OFF` and providing them by hand in CMAKE_CUDA_ARCHITECTURES is weird, we can rename USE_CMAKE_CUDA_ARCHITECTURES to GENERATE_CMAKE_CUDA_ARCHITECTURES.

Reply:
No sure how to handle  CMAKE_CUDA_ARCHTECTURES  without introduce more confusion, I have reverted the changes